### DISCUSSION TITLE
How to delete in cascade?

### DISCUSSION THREAD

**Discussion by pawamoy** [collerek/ormar#331] [Q&A] [answered]
I have this route:

```python
@router.delete("/{team_id}")
async def delete_team(team_id: int) -> None:
    team = await Team.objects.get(pk=team_id)
    await team.delete()
```

Other objects in the database have foreign keys pointing to teams.
When I call this route, I get this:

```json
{
  "error": "integrity error",
  "details": "FOREIGN KEY constraint failed"
}
```

The response is coming from this handler:

```python
@app.exception_handler(sqlite3.IntegrityError)
async def integrity_error_handler(request: Request, exc: sqlite3.IntegrityError) -> JSONResponse:
    return JSONResponse(
        status_code=400,
        content={"error": "integrity error", "details": str(exc)},
    )
```

I searched for "cascade" in the docs but couldn't find anything. Was I supposed to set `ondelete="cascade"` somewhere in the models?

**Comment by pawamoy** [comment]
Ah, I see, it's here: https://collerek.github.io/ormar/api/fields/foreign-key/#populate_fk_params_based_on_to_model
Still not sure what value I should put in `ondelete=...` 😕

**Comment by collerek** [comment, marked as answer]
Hmm maybe I missed that in the docs, you can pass `ondelete` and `onupdate` in `ForeignKey` like:

```python
source: Optional[DataSource] = ormar.ForeignKey(DataSource, ondelete="CASCADE")
```

If you want to cascades the delete then use.. `CASCADE` :) 
Those options are passed to sqlalchemy so you can check available options there.

**Reply by pawamoy** [reply]
Thanks !
A link to the accepted values on SQLAlchemy's docs would be very handy 😄

**Reply by collerek** [reply]
https://docs.sqlalchemy.org/en/14/core/constraints.html#on-update-and-on-delete :)
